exec.go

```diff
@@ -126,57 +125,50 @@ func setPrimaryKey(rv reflect.Value, id int64) {
 // the record in the DB with one Exec.
 // The given data needs to be:
 //
-// *[]*strcut
+// *[]*struct
 // *[]struct
 // []*struct
 // []struct
 //
 // sqlpro will executes one INSERT statement per call.
 func (db *DB) InsertBulk(table string, data interface{}) error {
-	var (
-		rv         reflect.Value
-		structMode bool
-		err        error
-	)
-
-	rv, structMode, err = checkData(data)
-	if err != nil {
+	rv, structMode, err := checkData(data)
+	switch {
+	case err != nil:
 		return err
-	}
-
-	if structMode {
+	// i don't know if we really want to fail silently
+	// returning back an error will probably be a good thing
+	// for the user
+	case rv.Len() == 0:
+		return nil
+	case structMode:
 		return fmt.Errorf("InsertBulk: Need Slice to insert bulk.")
 	}
 
-	key_map := make(map[string]*fieldInfo, 0)
-	rows := make([]map[string]interface{}, 0)
-
-	if rv.Len() == 0 {
-		return nil
-	}
+	key_map := make(map[string]*fieldInfo)
+	rows := make([]map[string]interface{}, rv.Len())
```

Comment:
Using "len" to initialize the slice is a good idea. But I think its even better to use "0,cap" for this, an below use "append" instead. The "len,cap" initialization needs to not only reserve the memory for the rows, but also to zero all members.